examples/plot_pp_study.py

```diff
@@ -0,0 +1,362 @@
+#
+# **Exploration: What if we attempt to apply Fairlearn's mitigation functionality?**
+#
+#
+# *Quickstart - Mitigating Disparity*
+#
+#
+# If we observe disparities between groups we may want to create a new
+# model while specifying an appropriate fairness constraint. Note that the
+# choice of fairness constraints is crucial for the resulting model, and
+# varies based on application context. If selection rate is highly
+# relevant for fairness in this contrived example, we can attempt to
+# mitigate the observed disparity using the corresponding fairness
+# constraint called Demographic Parity. In real world applications we need
+# to be mindful of the sociotechnical context when making such decisions.
+# The Exponentiated Gradient mitigation technique used fits the provided
+# classifier using Demographic Parity as the objective, leading to a
+# vastly reduced difference in selection rate.
+#
+# *Hypothesis*
+#
+#
+# In this sociotechnical context, the disparity is inherent in the data
+# itself, and thus an attempt at mitigation will fail.
+#
+np.random.seed(0)  # set seed for consistent results with ExponentiatedGradient
+constraint = DemographicParity()
+classifier = DecisionTreeClassifier(min_samples_leaf=10, max_depth=4)
+mitigator = ExponentiatedGradient(classifier, constraint)
+mitigator.fit(data_without_crimerate, y_true, sensitive_features=lstat_binned)
+y_pred_mitigated = mitigator.predict(data_without_crimerate)
+
+sr_mitigated = MetricFrame(selection_rate, y_true, y_pred_mitigated, sensitive_features=lstat_binned)
+print(sr_mitigated.overall)
+print(sr_mitigated.by_group)
+
+as_mitigated = MetricFrame(accuracy_score, y_true, y_pred_mitigated, sensitive_features=lstat_binned)
+print(as_mitigated.overall)
+print(as_mitigated.by_group)
+# %%
+#
+# By constraining selection rate disparity across LSTAT groups, we notice
+# a significant difference in the accuracy for the 3rd and 4th bins:
+#
+# **Before Mitigation - 95%+ accuracy on both bins**
+#
+#
+# **After Mitigation - 74% and 42% respectively**
+#
+#
+# The significant drop in accuracy indicates that the underlying issue may
+# not be at the model level but rather a reflection of the data (and
+# sociotechnical context itself). In the given data, LSTAT is a reliable
+# predictor of the crime rate threshold condition, which is a reflection
+# of systemic biases leading to this observed trend. This is why accuracy
+# for the higher bins falls off drastically if we try to achieve similar
+# selection rates across LSTAT bins.
+#
+# Problem Area & Sociotechnical Context
+# -------------------------------------
+#
+# The case study above highlights the importance of considering the
+# sociotechnical aspects of tasks traditionally deemed to be solely ML
+# problems. In this scenario, we’ve seen the grave consequences of failing
+# to acknowledge and ensure such considerations are a part of the design
+# and deployment of such ML systems. The result? A feedback loop of
```

Comment:
I understand what you mean, but I think we need to be much more explicit in pointing out what the feedback loop is here and how it gets created. We can't assume everyone has the background to infer it.

Reply:
Very much agree.